lib/widgets/lightbox.dart

```diff
@@ -0,0 +1,200 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:intl/intl.dart';
+
+import '../api/core.dart';
+import '../api/model/model.dart';
+import 'content.dart';
+import 'store.dart';
+
+class LightboxHero extends Hero {
+  LightboxHero({
+    super.key,
+    required Message message,
+    required String src,
+    required super.child
+  }) : super(
+    // TODO: Add index of the image preview in the message, to not break if
+    //   there are multiple image previews with the same URL in the same
+    //   message. Maybe keep `src`, so that on exit the lightbox image doesn't
+    //   fly to an image preview with a different URL, following a message edit
+    //   while the lightbox was open.
+    tag: '${message.id.toString()} $src'
+  );
+}
+
+class _Page extends StatefulWidget {
+  const _Page({
+    required this.routeEntranceAnimation,
+    required this.message,
+    required this.src
+  });
+
+  final Animation routeEntranceAnimation;
+  final Message message;
+  final String src;
+
+  @override
+  State<_Page> createState() => _PageState();
+}
+
+class _PageState extends State<_Page> {
+  // TODO: Animate entrance/exit of header and footer
+  bool _headerFooterVisible = false;
+
+  void _handleRouteEntranceAnimationStatusChange(AnimationStatus status) {
+    if (status == AnimationStatus.completed) {
+      setState(() {
+        _headerFooterVisible = true;
```

Comment:
I think so?

[`PageRouteBuilder.pageBuilder`](https://api.flutter.dev/flutter/widgets/PageRouteBuilder/pageBuilder.html) (where we consume the relevant `Animation<double>`)
https://github.com/flutter/flutter/blob/2ad6cd72c0/packages/flutter/lib/src/widgets/pages.dart#L94
```dart
  /// See [ModalRoute.buildPage] for complete definition of the parameters.
```

[`ModalRoute.buildPage`](https://api.flutter.dev/flutter/widgets/ModalRoute/buildPage.html)
https://github.com/flutter/flutter/blob/2ad6cd72c0/packages/flutter/lib/src/widgets/routes.dart#LL1062-L1064C41
```dart
  ///  * [animation]: The animation for this route's transition. When entering,
  ///    the animation runs forward from 0.0 to 1.0. When exiting, this animation
  ///    runs backwards from 1.0 to 0.0.
```

[`ModalRoute.buildPage`](https://api.flutter.dev/flutter/widgets/ModalRoute/buildPage.html)
https://github.com/flutter/flutter/blob/2ad6cd72c0/packages/flutter/lib/src/widgets/routes.dart#L1069-L1071
```dart
  /// This method is only called when the route is first built, and rarely
  /// thereafter. In particular, it is not automatically called again when the
  /// route's state changes unless it uses [ModalRoute.of]. […]
```

As long as `ModalRoute.buildPage` is not called synchronously with the beginning of the route exit, I think our listener will observe a transition from `completed` to something else ([`AnimationStatus.reverse`](https://api.flutter.dev/flutter/animation/AnimationStatus.html#reverse)?) at that time. (I say "as long as" because I think a fresh call to `ModalRoute.buildPage` would tear down our current listener and we'd make a new listener, and in theory the state change could happen synchronously in between, when we have no listener.)

Reply:
So maybe we add a case for `AnimationStatus.reverse` that sets `_headerFooterVisible` to false.

Reply:
> So maybe we add a case for `AnimationStatus.reverse` that sets `_headerFooterVisible` to false.

Sure, except let's make it comprehensive: if the status is completed, we want true, and if not completed, we want false. Right?